specification/peering/resource-manager/Microsoft.Peering/Peering/models.tsp

```diff
@@ -0,0 +1,1599 @@
+  IxRs: "IxRs",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  Voice: "Voice",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  EdgeZoneForOperators: "EdgeZoneForOperators",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  PeerProp: "PeerProp",
+}
+
+/**
+ * The prefix validation state.
+ */
+union PrefixValidationState {
+  string,
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  None: "None",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  Invalid: "Invalid",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  Verified: "Verified",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  Failed: "Failed",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  Pending: "Pending",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  Warning: "Warning",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  Unknown: "Unknown",
+}
+
+/**
+ * The prefix learned type
+ */
+union LearnedType {
+  string,
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  None: "None",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  ViaServiceProvider: "ViaServiceProvider",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required" "For backward compatibility"
+  ViaSession: "ViaSession",
+}
+
+/**
+ * The CDN peering prefix
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "For backward compatibility"
+model CdnPeeringPrefix extends Resource {
+  /**
+   * The properties that define a cdn peering prefix.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-private-usage" "For backward compatibility"
+  @Azure.ResourceManager.Private.conditionalClientFlatten
+  properties?: CdnPeeringPrefixProperties;
+}
+
+/**
+ * The properties that define a CDN peering prefix
+ */
+model CdnPeeringPrefixProperties {
+  /**
+   * The prefix.
+   */
+  @visibility(Lifecycle.Read)
+  prefix?: string;
+
+  /**
+   * The Azure region.
+   */
+  @visibility(Lifecycle.Read)
+  azureRegion?: string;
+
+  /**
+   * The Azure service.
+   */
+  @visibility(Lifecycle.Read)
+  azureService?: string;
+
+  /**
+   * The flag that indicates whether or not this is the primary region.
+   */
+  @visibility(Lifecycle.Read)
+  isPrimaryRegion?: boolean;
+
+  /**
+   * The BGP Community
+   */
+  @visibility(Lifecycle.Read)
+  bgpCommunity?: string;
+}
+
+/**
+ * The ARM resource class.
+ */
+model Resource {
```

Comment:
duplicate with common type.

Reply:
Same question, some of the models are still allof `Resource`, some are allof common type.

Reply:
.NET works now, thanks a lot